File: Python/EstruturaDeRepeticao/exercicio_49.py
"""
Faça um programa que mostre os n termos da Série a seguir:

  S = 1/1 + 2/3 + 3/5 + 4/7 + 5/9 + ... + n/m. 

Imprima no final a soma da série.
"""
def calcular_serie(n):
    soma = 0
    m = 1

    for i in range(1, n + 1):
        termo = i / m
        print(f"{i}/{m}", end=" ")
        soma += termo
        m += 2

    return soma

n_termos = int(input("Digite o número de termos da série: "))

soma_serie = calcular_serie(n_termos)
print(f"\nSoma da série: {soma_serie:.1f}")

File: Java/EstruturaDeRepeticao/Exercicio49.java
// Faça um programa que mostre os n termos da Série a seguir:
//   S = 1/1 + 2/3 + 3/5 + 4/7 + 5/9 + ... + n/m. 
// Imprima no final a soma da série.
package EstruturaDeRepeticao;
import java.util.Scanner;

public class Exercicio49 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite o valor de n: ");
        int n = scanner.nextInt();

        double soma = 0.0;
        int numerador = 1;
        int denominador = 1;

        System.out.print("Série: ");
        for (int i = 1; i <= n; i++) {
            if (i > 1) {
                System.out.print(" + ");
            }
            
            System.out.print(numerador + "/" + denominador);

            soma += (double) numerador / denominador;

            numerador++;
            denominador += 2;
        }

        System.out.println("\nSoma da série: " + soma);

        scanner.close();
    }
}
